[WPEPlatform] Properly obtain the EGLClientBuffer for an AHardwareBuffer

aperezdc opened the pull request:
#### bd2c0f76939390a3060fff96334531e211e3d89e
<pre>
[WPEPlatform] Properly obtain the EGLClientBuffer for an AHardwareBuffer
<a href="https://bugs.webkit.org/show_bug.cgi?id=302327">https://bugs.webkit.org/show_bug.cgi?id=302327</a>

Reviewed by Nikolas Zimmermann.

Fix EGLImage creation from AHardwareBuffer by properly using the
eglGetNativeClientBufferANDROID() function as required by Android EGL,
instead of directly casting AHardwareBuffer pointers to EGLClientBuffer.

This fixes the &quot;Failed to create EGLImage from AHardwareBuffer&quot; error
that prevents hardware-accelerated rendering on Android.

* Source/WebKit/WPEPlatform/wpe/WPEBufferAndroid.cpp:
(createImageEGL15): Receive EGLClientBuffer instead of AHardwareBuffer.
(createImageKHRImageBase): Ditto.
(wpeBufferAndroidImportToEGLImage): Obtain the EGLClientBuffer for the
AHardwareBuffer, and pass it instead to the helper functions used to
create the EGLImage for the buffer.

Co-authored-by: Felipe Erias &lt;felipeerias@igalia.com&gt;
Canonical link: <a href="https://commits.webkit.org/302857@main">https://commits.webkit.org/302857@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5cb7b9306b52ae56e4371e646c57e59539972633

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/130500 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/2771 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/41455 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/137918 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/82102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/155/builds/2780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/2663 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/99424 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/82102 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/133447 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/155/builds/2780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/116861 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/80123 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/155/builds/2780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81177 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/155/builds/2780 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/35496 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/140397 "Built successfully") | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/2561 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/2341 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/107937 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/2605 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/113203 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/107849 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27440 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/31644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/55539 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/158/builds/2631 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66019 "Built successfully") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/2450 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/2652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/152/builds/2557 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

nikolaszimmermann commented on Source/WebKit/WPEPlatform/wpe/WPEBufferAndroid.cpp:
```diff
@@ -129,7 +129,13 @@ static gpointer wpeBufferAndroidImportToEGLImage(WPEBuffer* buffer, GError** err
+    priv->eglImage = s_createImage(eglDisplay, clientBuffer);
     if (!priv->eglImage)
         g_set_error(error, WPE_BUFFER_ERROR, WPE_BUFFER_ERROR_IMPORT_FAILED, "Failed to import buffer to EGL image: eglCreateImageKHR failed with error %#04x", eglGetError());
     return priv->eglImage;
```
Drive-by review: EGL_NO_IMAGE? Instead of nullptr here too?

aperezdc replied:
Indeed, better to check against `EGL_NO_IMAGE` in the conditional, I'll change it before landing.

nikolaszimmermann reviewed (approved):
LGTM.

aperezdc committed:
[WPEPlatform] Properly obtain the EGLClientBuffer for an AHardwareBuffer
https://bugs.webkit.org/show_bug.cgi?id=302327

Reviewed by Nikolas Zimmermann.

Fix EGLImage creation from AHardwareBuffer by properly using the
eglGetNativeClientBufferANDROID() function as required by Android EGL,
instead of directly casting AHardwareBuffer pointers to EGLClientBuffer.

This fixes the "Failed to create EGLImage from AHardwareBuffer" error
that prevents hardware-accelerated rendering on Android.

* Source/WebKit/WPEPlatform/wpe/WPEBufferAndroid.cpp:
(createImageEGL15): Receive EGLClientBuffer instead of AHardwareBuffer.
(createImageKHRImageBase): Ditto.
(wpeBufferAndroidImportToEGLImage): Obtain the EGLClientBuffer for the
AHardwareBuffer, and pass it instead to the helper functions used to
create the EGLImage for the buffer.

Co-authored-by: Felipe Erias <felipeerias@igalia.com>
Canonical link: https://commits.webkit.org/302857@main
